### DISCUSSION TITLE
Use default constructor if no registration is found

### DISCUSSION THREAD

**Discussion by mykeels** [dadhi/DryIoc#639] [Q&A]
I have a controller:

```
public class MyController
{
  public MyController(): this(
    container.Resolve<MyService>()
  ) {}

  public MyController(MyService service)
  { ... }
}
```

When MyService is never registered, it rightly throws an UnableToResolveUnknownService exception.

However, MyService has a default constructor with empty parameters. Is there a way to get the DryIoc Container to use that when it finds no registrations for the concrete service class?

**Comment by dadhi** [comment]
@mykeels You may use the `.WithConcreteTypeDynamicRegistrations` described here https://github.com/dadhi/DryIoc/blob/master/docs/DryIoc.Docs/RulesAndDefaultConventions.md#withconcretetypedynamicregistrations
